### ISSUE TITLE
setTime not working correctly

### ISSUE THREAD

**Issue by codeflows** [date-fns/tz#24]
Hi,

I'm using date-fns (`date-fns 4.0.0, @date-fns/tz 1.1.2`) and noticed that e.g. `addMinutes(TZDate, number)` doesn't seem to work, at least if we look at the result of `toISOString()`.

As `addMinutes` [uses `TZDate.setTime` internally](https://github.com/date-fns/date-fns/blob/main/src/addMinutes/index.ts#L42), I tried to trace the bug and was able to come up with [this reproduction](https://github.com/codeflows/tz/commit/f024f749b7d4700c4b693f6b6b5629f4e5a53c0b). The test fails with:

```
 FAIL  src/date/tests.ts > TZDate > time > setTime > updated time is reflected in ISO timestamp
AssertionError: expected '1987-02-11T08:00:00.000+08:00' to deeply equal '2020-01-01T06:00:00.000+08:00'

- Expected
+ Received

- 2020-01-01T06:00:00.000+08:00
+ 1987-02-11T08:00:00.000+08:00

 ❯ src/date/tests.ts:337:36
    335|
    336|         date.setTime(+nativeDate);
    337|         expect(date.toISOString()).toEqual("2020-01-01T06:00:00.000+08:00");
       |                                    ^
    338|       });
    339|     });
```

On the other hand, helper functions such as `startOfDay(TZDate)` seem to work fine (`startOfDay` does `date.setHours(...)` internally).

**Comment by codeflows** [comment]
Duplicate of https://github.com/date-fns/tz/issues/16

**Comment by kossnocorp** [comment]
Thank you for reporting, I fixed and shipped it with `@date-fns/tz@1.2.0`
